packages/caver-transaction/src/transactionHelper/transactionHelper.js

```diff
@@ -0,0 +1,174 @@
+
+    TxTypeSmartContractDeploy: 'TxTypeSmartContractDeploy',
+    TxTypeFeeDelegatedSmartContractDeploy: 'TxTypeFeeDelegatedSmartContractDeploy',
+    TxTypeFeeDelegatedSmartContractDeployWithRatio: 'TxTypeFeeDelegatedSmartContractDeployWithRatio',
+
+    TxTypeSmartContractExecution: 'TxTypeSmartContractExecution',
+    TxTypeFeeDelegatedSmartContractExecution: 'TxTypeFeeDelegatedSmartContractExecution',
+    TxTypeFeeDelegatedSmartContractExecutionWithRatio: 'TxTypeFeeDelegatedSmartContractExecutionWithRatio',
+
+    TxTypeCancel: 'TxTypeCancel',
+    TxTypeFeeDelegatedCancel: 'TxTypeFeeDelegatedCancel',
+    TxTypeFeeDelegatedCancelWithRatio: 'TxTypeFeeDelegatedCancelWithRatio',
+
+    TxTypeChainDataAnchoring: 'TxTypeChainDataAnchoring',
+}
+
+const TX_TYPE_TAG = {
+    TxTypeLegacyTransaction: '',
+    '': TX_TYPE_STRING.TxTypeLegacyTransaction,
+
+    TxTypeValueTransfer: '0x08',
+    '0x08': TX_TYPE_STRING.TxTypeValueTransfer,
+    TxTypeFeeDelegatedValueTransfer: '0x09',
+    '0x09': TX_TYPE_STRING.TxTypeFeeDelegatedValueTransfer,
+    TxTypeFeeDelegatedValueTransferWithRatio: '0x0a',
+    '0x0a': TX_TYPE_STRING.TxTypeFeeDelegatedValueTransferWithRatio,
+
+    TxTypeValueTransferMemo: '0x10',
+    '0x10': TX_TYPE_STRING.TxTypeValueTransferMemo,
+    TxTypeFeeDelegatedValueTransferMemo: '0x11',
+    '0x11': TX_TYPE_STRING.TxTypeFeeDelegatedValueTransferMemo,
+    TxTypeFeeDelegatedValueTransferMemoWithRatio: '0x12',
+    '0x12': TX_TYPE_STRING.TxTypeFeeDelegatedValueTransferMemoWithRatio,
+
+    TxTypeAccountUpdate: '0x20',
+    '0x20': TX_TYPE_STRING.TxTypeAccountUpdate,
+    TxTypeFeeDelegatedAccountUpdate: '0x21',
+    '0x21': TX_TYPE_STRING.TxTypeFeeDelegatedAccountUpdate,
+    TxTypeFeeDelegatedAccountUpdateWithRatio: '0x22',
+    '0x22': TX_TYPE_STRING.TxTypeFeeDelegatedAccountUpdateWithRatio,
+
+    TxTypeSmartContractDeploy: '0x28',
+    '0x28': TX_TYPE_STRING.TxTypeSmartContractDeploy,
+    TxTypeFeeDelegatedSmartContractDeploy: '0x29',
+    '0x29': TX_TYPE_STRING.TxTypeFeeDelegatedSmartContractDeploy,
+    TxTypeFeeDelegatedSmartContractDeployWithRatio: '0x2a',
+    '0x2a': TX_TYPE_STRING.TxTypeFeeDelegatedSmartContractDeployWithRatio,
+
+    TxTypeSmartContractExecution: '0x30',
+    '0x30': TX_TYPE_STRING.TxTypeSmartContractExecution,
+    TxTypeFeeDelegatedSmartContractExecution: '0x31',
+    '0x31': TX_TYPE_STRING.TxTypeFeeDelegatedSmartContractExecution,
+    TxTypeFeeDelegatedSmartContractExecutionWithRatio: '0x32',
+    '0x32': TX_TYPE_STRING.TxTypeFeeDelegatedSmartContractExecutionWithRatio,
+
+    TxTypeCancel: '0x38',
+    '0x38': TX_TYPE_STRING.TxTypeCancel,
+    TxTypeFeeDelegatedCancel: '0x39',
+    '0x39': TX_TYPE_STRING.TxTypeFeeDelegatedCancel,
+    TxTypeFeeDelegatedCancelWithRatio: '0x3a',
+    '0x3a': TX_TYPE_STRING.TxTypeFeeDelegatedCancelWithRatio,
+
+    TxTypeChainDataAnchoring: '0x48',
+    '0x48': TX_TYPE_STRING.TxTypeChainDataAnchoring,
```

Comment:
We have fee-delegated chain data anchoring transactions as well.

Please see https://github.com/klaytn/klaytn/blob/dev/blockchain/types/tx_internal_data.go#L54.

Reply:
Thank you for letting me know about this ! I added :)